src/smt/smt_engine.cpp

```diff
@@ -3918,7 +3922,7 @@ void SmtEnginePrivate::processAssertions() {
     // proper data structure.
 
     // Placeholder for storing substitutions
-    d_assertions.getSubstitutionsIndex() = d_assertions.size();
+    d_preprocessingPassContext->getSubstitutionsIndex() = d_assertions.size();
```

Comment:
That is not really nice to read. Can't we add a `setSubstitutionIndex(...)` method?

Reply:
Done